build-logic/src/main/kotlin/polaris-java.gradle.kts

```diff
@@ -160,6 +160,45 @@ tasks.withType(Jar::class).configureEach {
   }
 }
 
+class StaleBehaviorChangeConfigurationChecker(private val appVersion: String) : FormatterFunc {
+
+  // Helper function for spotless to check for stale BehaviorChangeConfigurations
+  fun isStaleVersion(
+    annotatedVersion: String,
+    expiresVersion: String?,
+    polarisVersion: String,
+  ): Boolean {
+    val (majorA, minorA, patchA) = annotatedVersion.split(".").map { it.toInt() }
+    if (expiresVersion == null) {
+      val (majorB, minorB, _) = polarisVersion.split(".").map { it.toInt() }
+      return (majorB > majorA || (majorB == majorA && minorB > minorA + 1))
```

Comment:
Is it possible to have only on place for comparing versions, but derive the "expires" version from the "added" version automatically?

Reply:
I updated the PR to try this, but the logic might be less intuitive at first glance. Let me know what you think!